packages/tasks/src/model-libraries-snippets.ts

```diff
@@ -507,6 +507,32 @@ export const fastai = (model: ModelData): string[] => [
 learn = from_pretrained_fastai("${model.id}")`,
 ];
 
+export const sam2 = (model: ModelData): string[] => [
+	`import torch
+from sam2.sam2_image_predictor import SAM2ImagePredictor
+from sam2.sam2_video_predictor import SAM2VideoPredictor
+
+# for image prediction
+predictor = SAM2ImagePredictor.from_pretrained(${model.id})
+
+with torch.inference_mode(), torch.autocast("cuda", dtype=torch.bfloat16):
+    predictor.set_image(<your_image>)
+    masks, _, _ = predictor.predict(<input_prompts>)
+	
+# for video prediction
```

Comment:
Maybe return an array so we have individual snippets for image and video that can be copied by the user?

Reply:
we often put snippets to load them and not use them, no? this seems a bit overwhelming, the model cards have full examples already